src/ShortcodeAttsDynamicFunctionReturnTypeExtension.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/**
+ * Set return type of shortcode_atts().
+ */
+
+declare(strict_types=1);
+
+namespace PHPStan\WordPress;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\Type\Type;
+
+final class ShortcodeAttsDynamicFunctionReturnTypeExtension implements \PHPStan\Type\DynamicFunctionReturnTypeExtension
+{
+    public function isFunctionSupported(FunctionReflection $functionReflection): bool
+    {
+        return $functionReflection->getName() === 'shortcode_atts';
+    }
+
+    public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): Type
+    {
+        $argsCount = count($functionCall->args);
+        if ($argsCount === 0) {
```

Comment:
```suggestion
        if ($functionCall->args === []) {
```

Is it okay for you?

Reply:
Of course it is, you're the boss :) I'll update it later.

This is taken from another extension here. I just saw that the example in the phpstan docs do it without the assignment now, but you're way is fine too.